### PR TITLE
ISLAND-812 allow pdf update

### PR DESCRIPTION
**Pull request by pghorpade**

### CODE REVIEW

File: ucla_migration.drush.inc [hunk 1]
```diff
@@ -3821,14 +3821,42 @@ EOQ;
...
+            $object->relationships->add(FEDORA_MODEL_URI, 'hasModel', 'islandora:sp_pdf');
+            $datastream = $object->constructDatastream('OBJ', 'M');
+            $datastream->label = "PDF Document";
+            $datastream->mimetype = "application/pdf";
+            $is_new_pdf_ds = TRUE;
+        }
+
+        try {
+            $pdf_file = file_get_contents($pdf_path);
+            $datastream->content($pdf_file);
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: see [datastream methods](https://github.com/Islandora/islandora/wiki/Working-With-Fedora-Objects-Programmatically-Via-Tuque#methods-2) available with tuque
```suggestion
            $pdf_content = file_get_contents($pdf_path);
            $datastream->setContentFromString($pdf_content);
```

or 

```suggestion
            $datastream->setContentFromFile($pdf_path);
```

File: ucla_migration.drush.inc [hunk 2]
```diff
@@ -3821,14 +3821,42 @@ EOQ;
...
+                    '@label' => $object->label,
+                ),
+                WATCHDOG_NOTICE
+            );
+        }
+
+        watchdog('Ingest Success', 'Migrated object @file_name', array('@file_name' => $pdf_file), WATCHDOG_NOTICE);
     }
 
+
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: this is an unneeded line break

File: ucla_migration.drush.inc [hunk 3]
```diff
@@ -3821,14 +3821,42 @@ EOQ;
...
-        $object->ingestDatastream($datastream);
+
+        // get pdf datastream from object if exist or else create a new one add this logic
+        watchdog('Ingest Success', 'In add/update pdf @file_name', array('@file_name' => $pdf_path), WATCHDOG_NOTICE);
+        $is_new_pdf_ds = FALSE;
+        if (isset($object['OBJ'])) {
+            $datastream = $object['OBJ'];
+        }
+        else {
+            $object->relationships->add(FEDORA_MODEL_URI, 'hasModel', 'islandora:sp_pdf');
```
Comment (IAMlKeno) [inline ucla_migration.drush.inc]: is this necessary? If the object already exists, it more than likely _already has_ the islandora:sp_pdf model

You can use the [relationships->get()](https://github.com/Islandora/islandora/wiki/Working-With-Fedora-Objects-Programmatically-Via-Tuque#example-of-retrieving-a-filtered-relationship) method to check if this is present or not

File: ucla_migration.drush.inc (hunk 2 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: This is removed.

File: ucla_migration.drush.inc (hunk 3 above)
Comment (pghorpade) [reply ucla_migration.drush.inc]: Added a relationship check.

**Pull request merged**